drivers/graphtest/testutil.go

```diff
@@ -351,3 +357,14 @@ func readDir(dir string) ([]os.DirEntry, error) {
 
 	return b, nil
 }
+
+// removeLayer tries to remove the layer, retrying several times if an EBUSY
+// error is returned
+func removeLayer(t testing.TB, driver graphdriver.Driver, name string) {
+	err := driver.Remove(name)
+	for i := 100 * time.Millisecond; i <= 1000*time.Millisecond && err != nil && errors.Is(err, syscall.EBUSY); i += (100 * time.Millisecond) {
```

Comment:
(I’m totally unfamiliar with this) BTW is it understood why retries are necessary? I’d naively expect the tests to be one of the more deterministic environments with no unexpected concurrent accessors, so what is keeping the volume busy if it’s not the tests?

Are the tests somehow triggering hard-to-cleanup concurrent goroutines/threads/processes that they don’t wait for, maybe?

Reply:
And if it is understood, it might be worth documenting somewhere around here.

Reply:
There’s enough mentions of `EBUSY` throughout the codebase that I totally believe that there’s _something_ (or multiple somethings) going on — I just could never find any specific description of what is actually the reason. The closest is https://github.com/containers/storage/pull/719 / https://github.com/moby/moby/pull/36438 , which also isn’t really specific (and seems to suggest that the error is a genuine external user, not something that could be successfully retried).

So I’m mostly trying to get as much of the knowledge, if any, documented and explicit.
If this is working as designed and just requires a retry for good reasons, great, let’s document that.
If we know what is not working as designed, but needs a workaround short-term, fine, let’s document that.
If the cause is unknown but we know that something is failing, oh well, documenting that would still be nice.

If we are lucky, somebody will, sometime, find and fix the cause, and then look at this kind of loop and wonder whether it can be removed — and documenting why it was added would have been helpful.